presto-main/src/main/java/com/facebook/presto/operator/WindowOperator.java

```diff
@@ -567,12 +567,14 @@ void updateMemoryUsage()
         final SpillerFactory spillerFactory;
         final PageWithPositionComparator pageWithPositionComparator;
 
+        boolean spillingWhenConvertingRevocableMemory;
         boolean resetPagesIndex;
         int pendingInputPosition;
 
         Optional<Page> currentSpillGroupRowPage;
         Optional<Spiller> spiller;
-        ListenableFuture<?> spillInProgress = immediateFuture(null);
+        // Spill can be trigger by Driver, by us or both. `spillInProgress` is not empty when spill was triggered but not `finishMemoryRevoke()` yet
```

Comment:
I assume "us" here means this window operator, especially `SpillablePagesToPagesIndexes#process`